reconcile/openshift_rhcs_certs.py

```diff
@@ -224,17 +229,16 @@ def fetch_desired_state(
 ) -> None:
     vault = VaultClient()
     cert_provider = get_rhcs_provider_settings(query_func=query_func)
-
     for ns in namespaces:
         for cert_resource in ns.openshift_resources or []:
-            if isinstance(cert_resource, NamespaceOpenshiftResourceRhcsCertV1):
+            if _is_rhcs_cert(cert_resource):
                 ri.add_desired_resource(
                     cluster=ns.cluster.name,
                     namespace=ns.name,
                     resource=fetch_openshift_resource_for_cert_resource(
                         dry_run,
                         ns,
-                        cert_resource,
+                        cast("NamespaceOpenshiftResourceRhcsCertV1", cert_resource),
```

Comment:
`fetch_openshift_resource_for_cert_resource` should handle both `NamespaceOpenshiftResourceRhcsCertV1` and `SharedResourcesV1_NamespaceOpenshiftResourceV1_NamespaceOpenshiftResourceRhcsCertV1`